CryptoBinPkg/Plugin/ReportCrypto/ReportCrypto.py

```diff
@@ -0,0 +1,202 @@
+## @file BundleCrypto.py
+# Plugin to Bundle the CryptoBin build output
+#
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+from edk2toolext.environment.plugintypes.uefi_build_plugin import IUefiBuildPlugin
+from pathlib import Path
+from datetime import datetime
+from git import Repo
+import lzma
+
+# Dictionaries
+binary_to_type = {}
+arch_and_type_to_lib = {}
+
+# defines
+offset = 40
+
+class ReportCrypto(IUefiBuildPlugin):
+    """
+    ReportCrypto is a plugin class that generates a report about the crypto binaries built during the build process.
+    The report includes the next information:
+    - Submodules information
+    - Built binaries sizes
+    - Linked OpenSSL library for each binary
+    - OpenSSL library flags
+    Methods:
+    --------
+    do_post_build(thebuilder):
+        Generates a report file in the build directory containing information about the build process, 
+        submodules, built binaries, and linked OpenSSL configuration.
+    get_openssl_flags(file):
+        Extracts OpenSSL library flags from the given .inf file.
+    get_module_type_for_crypto_bin():
+        Retrieves the module type (DXE_DRIVER, PEIM, etc.) for each crypto binary from the .inf files in the CryptoBinPkg Driver path.
+    get_linked_lib(arch, module_type, lib):
+        Determines the linked library configuration for the specified architecture and module type from the CryptoBinPkg.dsc file.
+    """
+
+    def do_post_build(self, thebuilder):
+        ### Initiazlize variables ###
+
+        # get needed paths
+        openssl_lib_path = Path(thebuilder.edk2path.GetAbsolutePathOnThisSystemFromEdk2RelativePath("OpensslPkg")) / "Library" / "OpensslLib"
+        CryptoBinPkg_Driver_path = Path(thebuilder.ws) / "CryptoBinPkg" / "Driver"
+        cryptoBinPkg_dsc_path = Path(thebuilder.ws) / "CryptoBinPkg" / "CryptoBinPkg.dsc"
+        repo = Repo(thebuilder.ws)
+
+        # Path to Build output
+        build_path = Path(thebuilder.env.GetValue("BUILD_OUTPUT_BASE"))
+        tool_chain = thebuilder.env.GetValue("TOOL_CHAIN_TAG")
+        target = thebuilder.env.GetValue("TARGET")
+        # 
+        # Create report file in the build directory
+        report_file_name = f"Report_{thebuilder.flavor}_{target}_{tool_chain}.txt"
+        report_File = Path(thebuilder.ws) / "Build" / report_file_name
+        report_File.touch()
+        arch_list = thebuilder.env.GetValue("TARGET_ARCH")
+
+        # Time and Title
+        title = f"CRYPTO BINARIES REPORT - {thebuilder.flavor} FLAVOR - {target}\n\n"
+        time = datetime.now().strftime('%Y-%m-%d %H:%M:%S\n')
+        report = [title, "Build Time: " + time]
+
+        # get tool chain

```

Comment:
nit: A lot of these comments are unnecessary. If you need a comment for one or two lines of code, you need to make your code more readable, not add a line. I'll leave it up to you to clean up some of the comments.